### ISSUE TITLE
Custom Cloud Provider Doesn't Work

### ISSUE THREAD

**Issue by lots0logs** [rancher/terraform-provider-rke#196]
Currently the schema for `custom_cloud_provider.name` forces the name to be one of the built-in provider names or `custom`. That makes cluster creation fail with the following error:
```
error building controller context: cloud provider could not be initialized: unknown cloud provider "custom"
```
~~In my case, the name is `digitalocean`~~ Actually looks like it must be set to `external` but Terraform won't allow me to use it because it doesn't pass the current schema validation. 

References: 
* https://rancher.com/docs/rke/latest/en/config-options/cloud-providers/custom/
* https://github.com/rancher/terraform-provider-rke/issues/21

**Comment by WebSpider** [comment]
I just pass this as an extra_args to the kubelet service, and it works just fine for me (running on hetzner cloud)

**Comment by rawmind0** [comment]
Thanks for reporting. PR #197 should fix the issue

**Comment by rawmind0** [comment]
PR #197 merged on master. It will be included on release v1.1.1

Please reopen issue if needed.
